tests/src/test/scala/whisk/core/limits/ActionLimitsTests.scala

```diff
@@ -63,37 +60,154 @@ class ActionLimitsTests extends TestHelpers with WskTestHelpers {
+      case Some(TimeLimit.MAX_DURATION)            => s"${TimeLimit.MAX_DURATION} (= max)"
+      case Some(t) if (t < TimeLimit.MIN_DURATION) => s"${t} (< min)"
+      case Some(t) if (t > TimeLimit.MAX_DURATION) => s"${t} (> max)"
+      case Some(t)                                 => s"${t} (allowed)"
+    }
+
+    val toMemoryString = memory match {
+      case None                                   => "None"
+      case Some(MemoryLimit.minMemory)            => s"${MemoryLimit.minMemory.toMB.MB} (= min)"
+      case Some(MemoryLimit.stdMemory)            => s"${MemoryLimit.stdMemory.toMB.MB} (= std)"
+      case Some(MemoryLimit.maxMemory)            => s"${MemoryLimit.maxMemory.toMB.MB} (= max)"
+      case Some(m) if (m < MemoryLimit.minMemory) => s"${m.toMB.MB} (< min)"
+      case Some(m) if (m > MemoryLimit.maxMemory) => s"${m.toMB.MB} (> max)"
+      case Some(m)                                => s"${m.toMB.MB} (allowed)"
+    }
+
+    val toLogsString = logs match {
+      case None                                  => "None"
+      case Some(LogLimit.MIN_LOGSIZE)            => s"${LogLimit.MIN_LOGSIZE} (= min)"
+      case Some(LogLimit.STD_LOGSIZE)            => s"${LogLimit.STD_LOGSIZE} (= std)"
+      case Some(LogLimit.MAX_LOGSIZE)            => s"${LogLimit.MAX_LOGSIZE} (= max)"
+      case Some(l) if (l < LogLimit.MIN_LOGSIZE) => s"${l} (< min)"
+      case Some(l) if (l > LogLimit.MAX_LOGSIZE) => s"${l} (> max)"
+      case Some(l)                               => s"${l} (allowed)"
+    }
+
+    val toExpectedResultString: String = if (ec == SUCCESS_EXIT) "allow" else "reject"
+  }
+
+  val perms = (// Assert for valid permutations that the values are set correctly
+  for {
+    time <- Seq(None, Some(TimeLimit.MIN_DURATION), Some(TimeLimit.MAX_DURATION))
+    mem <- Seq(None, Some(MemoryLimit.minMemory), Some(MemoryLimit.maxMemory))
+    log <- Seq(None, Some(LogLimit.MIN_LOGSIZE), Some(LogLimit.MAX_LOGSIZE))
+  } yield PermutationTestParameter(time, mem, log)) ++
+    // Add variations for negative tests
+    Seq(
+      PermutationTestParameter(Some(0.milliseconds), None, None, BAD_REQUEST), // timeout that is lower than allowed
+      PermutationTestParameter(Some(TimeLimit.MAX_DURATION.plus(1 second)), None, None, BAD_REQUEST), // timeout that is slightly higher than allowed
+      PermutationTestParameter(Some(100.minutes), None, None, BAD_REQUEST), // timeout that is much higher than allowed
+      PermutationTestParameter(None, Some(0.MB), None, BAD_REQUEST), // memory limit that is lower than allowed
+      PermutationTestParameter(None, Some(MemoryLimit.maxMemory + 1.MB), None, BAD_REQUEST), // memory limit that is slightly higher than allowed
+      PermutationTestParameter(None, Some(32768.MB), None, BAD_REQUEST), // memory limit that is much higher than allowed
+      PermutationTestParameter(None, None, Some(32768.MB), BAD_REQUEST)) // log size limit that is much higher than allowed
```

Comment:
Shall we make all the limits relative to the respective min/max values instead of using magic numbers?